framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java

```diff
@@ -1816,6 +1815,18 @@ private void annotateTypeParameters(
                     stubWarnNotFound(
                             param, "Annotations on intersection types are not yet supported");
                 }
+                if (param.getTypeBound().size() == 1
+                        && param.getTypeBound().get(0).getAnnotations().isEmpty()
+                        && paramType
+                                .getUpperBound()
+                                .getUnderlyingType()
+                                .toString()
+                                .contentEquals("java.lang.Object")) {
+                    // If there is an explicit "T extends Object" type parameter bound,
+                    // treat it like an explicit use of "Object" in code.
+                    AnnotatedTypeMirror ub = atypeFactory.getAnnotatedType(Object.class);
```

Comment:
In theory, it seems like we'd want to apply any defaults associated with this location, like those set by `@DefaultQualifier`. In practice, that probably doesn't come up. And it sounds potentially hard to do :)

Reply:
I was also a bit worried about that, but at the moment I don't see anything that would suggest a `@DefaultQualifier` in an astub file (and therefore in the annotated JDK) has any effect.
That would be another unfortunate mismatch between `.java` files and `.astub` files, but is probably hard to fix. Let's punt on that until we actually run into that issue.

Reply:
(We've actually used `@DefaultQualifier` in a stub file successfully but only with the goal of affecting the defaults that the checker applies to the actual source code. I don't know that we've ever tried to use it to affect what's written in the stub file, which is what we're talking about here.)